presto-parquet/src/main/java/io/prestosql/parquet/GroupField.java

```diff
@@ -19,17 +19,25 @@
 import java.util.List;
 import java.util.Optional;
 
+import static com.google.common.base.Preconditions.checkArgument;
 import static java.util.Objects.requireNonNull;
 
 public class GroupField
         extends Field
 {
-    private final ImmutableList<Optional<Field>> children;
+    private final List<Optional<Field>> children;
 
-    public GroupField(Type type, int repetitionLevel, int definitionLevel, boolean required, ImmutableList<Optional<Field>> children)
+    public GroupField(Type type, int repetitionLevel, int definitionLevel, boolean required, List<Optional<Field>> children)
     {
         super(type, repetitionLevel, definitionLevel, required);
-        this.children = requireNonNull(children, "children is required");
+        checkArgument(
```

Comment:
Is there a test for this assertion?

Reply:
There isn't. Users of `GroupField` were silently assuming this.

Reply:
We probably should have a test for this as for invalid decimal types

Reply:
Today it's statically provable that the condition checked here always holds
https://github.com/prestosql/presto/blob/3664123b1e5b3c01fb3e7b3385fab59c288143a3/presto-hive/src/main/java/io/prestosql/plugin/hive/parquet/ParquetColumnIOConverter.java#L54-L91

This means, there isn't a reasonable test I could write to cover this.
The check is still useful, because it helps _users_ of the class understand what's going on.